FIRAuthWebUtilsTests.m

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <XCTest/XCTest.h>
+#import "FirebaseAuth/Sources/Utilities/FIRAuthWebUtils.m"
+
+/** @class FIRAuthWebUtilsTests
+ @brief Tests for the FIRAuthWebUtils class.
+ */
+@interface FIRAuthWebUtilsTests : XCTestCase
+@end
+
+@implementation FIRAuthWebUtilsTests
```

Comment:
Add a failure case?

Reply:
As discussed offline, this function just parses domain from url and doesn't throw an error. Url validation happens on redirection on the backend, we can add SDK validation later as needed.